src/main/java/ContactInfoPage.java

```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
         super(driver);
     }
 
+    @FindBy(id="btn-edit-contact")
+    private WebElement editButton;
 
-    @FindBy(css = "button.btn.btn-primary")
+    @FindBy(id="contact-first-name")
+    private WebElement firstNameInputField;
+
+    @FindBy(id="contact-last-name")
+    private WebElement lastNameInputField;
+
+    @FindBy(id="contact-description")
+    private WebElement contactDescription;
+
+    @FindBy(css= "[class=\"btn btn-primary submit-btn-ec\"]")
     private WebElement saveButton;
-    @FindBy(css = "h5")
-    private WebElement contactInformationHeader;
-    @FindBy(css = "a.nav-link[routerlink='']")
-    private WebElement contactsList;
-    @FindBy(css = "input[formcontrolname='searchInput']\"")
-    private WebElement searchInputForm;
-    @FindBy(css = "b")
-    private WebElement foundContact;
-    @FindBy(css = "img[src='/assets/icons/trash.svg']")
-    private WebElement deleteButton;
+
+    @FindBy(xpath = "//*[@id=\"edit-contact-form\"]/div[1]/div[2]/input")
+    private WebElement firstNameEdit;
+
+    @FindBy(xpath = "//*[@id=\"edit-contact-form\"]/div[2]/div[2]/input")
+    private WebElement lastNameEdit;
+
+    @FindBy(xpath = "//*[@id=\"edit-contact-form\"]/div[3]/div[2]/textarea")
+    private WebElement descriptionEdit;
+
+    @FindBy(xpath = "/html/body/app-root/app-home-page/app-header/nav/div/ul/li[1]/a")
+    private WebElement contactButton;
+
+
+
+
+    public String firstNameIsCorrect(){
+        return firstNameInputField.getText();
+    }
+
+    public String lastNameIsCorrect(){
+        return lastNameInputField.getText();
+    }
+
+    public String descriptionIsCorrect(){
+        return contactDescription.getText();
+    }
+
+
+    public void clickButtonEdit(){
+        try{
+            editButton.click();
+        } catch (ElementClickInterceptedException e){
+            e.getStackTrace();
+        }
+
+    }
+
+    public void enterFirstNameEdit(String firstNameEditValue){
```

Comment:
I think so better: editFirstName

Reply:
In addition, when editing fields, it is necessary to delete the old data in the field.

For example,

`editFirstName.clear();`

Reply:
I fix it